src/capsule/BiometricSessionManager.ts

```diff
@@ -0,0 +1,42 @@
+import userManagementClient from './UserManagementClient'
+import { ChallengeStorage } from './ChallengeStorage'
+
+export default class BiometricSessionManager {
+  private userId: string
+  private biometricStorage: ChallengeStorage
+  public async setBiometrics() {
+    return await userManagementClient.addBiometrics(this.userId, {
+      publicKey: await this.biometricStorage.getPublicKey(),
+    })
+  }
+
+  constructor(userId: string, biometricStorage: ChallengeStorage) {
+    this.userId = userId
+    this.biometricStorage = biometricStorage
+  }
+
+  private cookie: string | undefined
+
+  public async refreshBiometricsIfNeeded() {
+    if (typeof this.cookie === 'string') {
+      // this is how cookie is represented. We do parsing "manually" to avoid employing additional libs
+      // Example cookie: capsule.sid=s%3Ad324cb79-96c8-4995-868b-4774ae2004ce.RZ2H%2BbendbOVXEBJ2tKVLatSh24SOxxQ%2F7A51lfdSoM; Path=/; Expires=Fri, 30 Dec 2022 18:31:47 GMT; HttpOnly; SameSite=Strict
+      const expDate = this.cookie
```

Comment:
I don't follow this syntax. For example, shouldn't it be `split(';')` instead of `split?.(';')`. `split` will always exist as a method as long as the string before it is defined so there's no need to add `?` and there should be no `.` before the parenthesis, right?

Reply:
It also looks like we need more error handling (ex. index out of bounds error on [1]). It might be cleaner to do a regex check and throw an error if it doesn't find a match. WDYT?

Reply:
```
const expMatch = this.cookie.match(/Expires=(.*?);/)
if(!expMatch || expMatch.length < 1) {
  // error
}
const expDate = expMatch[1]
```